tests/test_squeues.py

```diff
@@ -114,7 +114,7 @@ def test_nonserializable_object(self):
         # Trigger Twisted bug #7989
         import twisted.persisted.styles  # NOQA
         q = self.queue()
-        self.assertRaises(ValueError, q.push, lambda x: x)
+        self.assertRaises((ValueError, AttributeError), q.push, lambda x: x)
```

Comment:
This can break user code if it expects ValueError from MarshalLifoDiskQueue (and friends). Do you know how Scrapy handles these errors itself? It should be catching serialization errors somewhere to decide whether put a Request to a disk queue or to in-memory queue.